site/_includes/partials/post-navigation.njk

```diff
@@ -0,0 +1,33 @@
+{% set contentItems = null %}
+{% set contentType = "" %}
+
+{% if page.url | getContentType === 'articles' %}
+  {% set contentItems = collections.articles %}
+{% endif %}
+
+{% if page.url | getContentType === 'blog' %}
+  {% set contentItems = collections.blogPosts %}
+{% endif %}
+
+{% set currentPostIndex = contentItems | indexOfPostByUrl(page.url) %}
+{% set previousPost = contentItems[currentPostIndex - 1] %}
+{% set nextPost = contentItems[currentPostIndex + 1] %}
```

Comment:
There is https://www.11ty.dev/docs/filters/collection-items/#getpreviouscollectionitem-and-getnextcollectionitem - is that of help here?